### PR TITLE
connection to mqtt broker and subscribing to events

### PR DESCRIPTION
**Pull request by surjak**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        TAKE_PLACE("take"), LEAVE_PLACE("leave");
+
+        fun getName(): String {
+            return this.str;
+        }
+
+        companion object {
+            fun fromStr(str: String): ParkingPlaceEvent {
+                return if (str == TAKE_PLACE.getName()) TAKE_PLACE else if (str == LEAVE_PLACE.getName()) LEAVE_PLACE else throw IllegalArgumentException()
+            }
```
Comment (HubertBalcerzak) [inline src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt:52]:
```suggestion
fun fromStr(str: String): ParkingPlaceEvent = when (str) {
                TAKE_PLACE.name -> TAKE_PLACE
                LEAVE_PLACE.name -> LEAVE_PLACE
                else -> throw IllegalArgumentException()
            }
```

File: src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+
+    enum class ParkingPlaceEvent(val str: String) {
+        TAKE_PLACE("take"), LEAVE_PLACE("leave");
+
+        fun getName(): String {
+            return this.str;
+        }
+
```
Comment (HubertBalcerzak) [inline src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt:48]:
```suggestion
```

File: src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            }
+
+        }
+
+        private fun tryParse(topic: String, message: String): MqttCupCarbonMessage {
+            val list = topic.split("/").toList()
+            val parkingPlaceEvent = ParkingPlaceEvent.fromStr(list[1])
+            val floor = toInt(list[2])
+            val parkingPlace = toInt(list[3])
+            val carId = toInt(message)
```
Comment (HubertBalcerzak) [inline src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt:33]:
```suggestion
            val floor = list[2].toInt()
            val parkingPlace = list[3].toInt()
            val carId = message.toInt()
```

File: src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            return MqttCupCarbonMessage(parkingPlaceEvent, floor, parkingPlace, carId)
+        }
+
+        private fun toInt(str: String): Int {
+            return Integer.valueOf(str)
+        }
```
Comment (HubertBalcerzak) [inline src/main/kotlin/pl/edu/agh/smartparking/mqtt/MqttCupCarbonMessage.kt:39]:
```suggestion
```

**Review by HubertBalcerzak — CHANGES_REQUESTED** [review]

**Comment by surjak** [comment]
Poprawione 😃

**Pull request merged**